pkg/splunk/enterprise/util.go

```diff
@@ -949,6 +950,45 @@ func GetNextRequeueTime(appRepoPollInterval, lastCheckTime int64) time.Duration
 	return time.Second * (time.Duration(nextRequeueTimeInSec))
 }
 
+// isAppRepoPollingEnabled checks whether automatic polling for apps repo changes
+// is enabled or not. If the value is 0, then we fallback to on-demand polling of apps
+// repo changes.
+func isAppRepoPollingEnabled(appStatusContext *enterpriseApi.AppDeploymentContext) bool {
+	return appStatusContext.AppsRepoStatusPollInterval != 0
+}
+
+func shouldCheckAppRepoStatus(client splcommon.ControllerClient, cr splcommon.MetaObject, appFrameworkConf *enterpriseApi.AppFrameworkSpec, appStatusContext *enterpriseApi.AppDeploymentContext, kind string, turnOffManualChecking *bool) bool {
+	scopedLog := log.WithName("shouldCheckAppRepoStatus").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace())
+	// Do not poll for the app updates if the manual app update is turned off.
+	if !reflect.DeepEqual(appStatusContext.AppFrameworkConfig, *appFrameworkConf) {
```

Comment:
Now that you have said it, I thought about it and now I know why this needs to be reverted. I made this change to not poll for apps during the first time, but what this will do is it will let the init container download the apps but will not install it. And even if we change the status to `on`, we wont install the apps since pod is not recycled and hence ansible playbooks are not run to install the apps.